sympy/integrals/prde.py

```diff
@@ -441,6 +441,50 @@ def prde_no_cancel_b_small(b, Q, n, DE):
     return f + H, A.col_join(B).col_join(C)
 
 
+def prde_cancel_liouvillian(b, Q, n, DE):
+    """
+    Pg, 237.
+    Examples of this are:
+    log(x + 1)*log(x), log(x + 1)/log(x)
+
+    """
+    m = len(Q)
+    f, A = [None]*(n + 1), [None]*(n + 1)
+    #H = [[Poly(0, DE.t)]*m for i in range(n + 1)]
+
+    # Why use DecrementLevel? Below line answers that:
+    # Assuming that we can solve such problems over 'k' (not k[t])
+    if DE.case == 'primitive':
+        with DecrementLevel(DE):
+            ba, bd = frac_in(b, DE.t, field=True)
+    elif DE.case == 'exp':
+        with DecrementLevel(DE):
+            ba, bd = frac_in(b + n*derivation(DE.t, DE)/DE.t, DE.t, field=True)
+
+    for N in range(n, -1, -1):
+        with DecrementLevel(DE):
+            Qyn = [frac_in(q.nth(N), DE.t) for q in Q]
+            fn, An = param_rischDE(ba, bd, Qyn, DE)
+        fn = [Poly(fa.as_expr()/fd.as_expr(), DE.t, field=True)
+                for fa, fd in fn]
+
+        V = An.nullspace()
+        # take the first vector of nullspace
+        # v = [c1, ..., cm, d1n, ..., drn_n]
+        v = An[0]
```

Comment:
It is too early to pick a solution. All options should be left available until the solutions of param_rischDE have been found for all powers of t: `t**n, ..., t**i, ..., t**0`. Then the results are combined to a big matrix. which is returned together with a list of solution components `h_ji = f_ji t**i`.